### Annotation

Annotation simply means the process of adding explanatory notes or comment, like highlighting certain lines of your important chapter or underlining interesting lines of your notebook.

The sequence that play with in bioinformatics, holds various information at different regions. The prediction of the function of gene or a region of chromosome by analyzing raw sequence is called genome annotation.

Location of gene is identified
Coding regions are identified
Start and Stop points are identified
Function of gene is identified

Annotation is a predictive activities and confirmation of those predictions is done through functional genomics

# Prokka
Prokka is a software tool to annotate bacterial, archaeal and viral genomes quickly and produce standards-compliant output files in different formats like .gff, .tsv, .faa, etc

## From their FAQ

Where does the name "Prokka" come from?

Prokka is a contraction of "prokaryotic annotation". It's also relatively unique within Google, and also rhymes with a native Australian marsupial called the quokka.



Please refer: 

https://academic.oup.com/bioinformatics/article/30/14/2068/2390517

https://github.com/tseemann/prokka

Today, we will be following a youtube tutorial by Bioinformatics Coach to get introduced with Prokka.

Here is the youtube link which I have followed for annotation of complete sequence of Mycobacterium ulcerans Agy99:

https://youtu.be/NYbYxGf-8NM


<iframe width="1350" height="480" src="https://www.youtube.com/embed/NYbYxGf-8NM" title="A bioinformatics tutorial on genome annotation of bacterial genomes using PROKKA (updated video)" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

Let's get started by downloading Fasta sequence from : https://www.ncbi.nlm.nih.gov/nuccore/NC_008611.1

I have downloaded the fasta file as sequence.fasta and opened terminal on same directory.

In [ ]:
prokka --cpus 4 --prefix Agy99 --kingdom Bacteria --locustag Agy99 sequence.fasta

In [1]:
ls # listing files after running Prokka

Agy99  sequence.fasta  Untitled.ipynb


In [3]:
tree # # viewing files list in tree form

.
├── Agy99
│   ├── Agy99.err
│   ├── Agy99.faa
│   ├── Agy99.ffn
│   ├── Agy99.fna
│   ├── Agy99.fsa
│   ├── Agy99.gbk
│   ├── Agy99.gff
│   ├── Agy99.log
│   ├── Agy99.sqn
│   ├── Agy99.tbl
│   ├── Agy99.tsv
│   └── Agy99.txt
├── sequence.fasta
└── Untitled.ipynb

1 directory, 14 files


### .txt file

In [14]:
cat Agy99/Agy99.txt  # concatenating Agy99.txt file

organism: Genus species strain 
contigs: 1
bases: 5631606
CDS: 5471
rRNA: 3
tRNA: 48
tmRNA: 1


###### informations like number of contigs, bases, number of genes for coding sequences,rRNA,tRNA,etc is obtained

### .faa file

###### .faa files give predicted sequences of protein

In [30]:
head Agy99/Agy99.faa # getting first 10 lines

>Agy99_00001 Chromosomal replication initiator protein DnaA
MWNAVVAELNGEPNTDGDAGTGTTLTSPLTPQQRAWLNLVQPLTIVEGFALLSVPSSFVQ
NEIERHLRTPITAALSRRLGQQIQLGVRIAPPPADDDDDSVAAAVEDPGLEASPETSQEV
SDEIDDFGENAPKSRQSWPTHFKKRSTDADTSASADGTSLNRRYTFDTFVIGASNRFAHA
ATLAIAEAPARAYNPLFIWGESGLGKTHLLHAAGNYAQRLFPGMRVKYVSTEEFTNDFIN
SLRDDRKVAFKRSYRDVDVLLVDDIQFIEGKEGIQEEFFHTFNTLHNANKQIVISSDRPP
KQLATLEDRLRTRFEWGLITDVQPPELETRIAILRKKAQMERLAVPDDVLELIASSIERN
IRELEGALIRVTAFASLNKTPIDKALAEIVLRDLIADADTMQISAATIMAATVEYFDTTV
EELRGPGKTRALAQSRQIAMYLCRELTDLSLPKIGQAFGRDHTTVMYAQRKILSEMAERR
EVFDHVKELTTRIRQRSKR


#### Narrative: Chromosomal replication initiator protein DnaA is predicted with above sequence

In [32]:
# counting number of fasta sequences in the file
grep '>' Agy99/Agy99.faa | wc -l 

5471


In [9]:
# looking for specific names
grep 'hypothetical' Agy99/Agy99.faa | wc -l

2516


#### Narrative: 2516 potential genes were found but while comparing those gene on the database no matches were present.
##### Potential genes means Probable gene that are making unknown proteins

### .ffn file

###### .ffn files give predicted sequences of nucleotides

In [11]:
grep '>' Agy99/Agy99.ffn | wc -l

5523


#### Narrative: 5523 sequences are present

In [34]:
head Agy99/Agy99.ffn # getting first 10 lines

>Agy99_00001 Chromosomal replication initiator protein DnaA
GTGTGGAATGCCGTCGTCGCTGAGCTCAACGGAGAACCCAACACCGATGGTGATGCCGGC
ACCGGCACCACTTTGACCAGTCCACTAACCCCCCAGCAAAGAGCTTGGTTGAACCTCGTC
CAACCGCTCACCATCGTTGAGGGGTTTGCTCTGCTGTCGGTCCCGAGCAGCTTTGTGCAA
AACGAGATCGAACGCCACCTGCGCACCCCGATCACCGCAGCGCTGAGCCGTCGGCTTGGA
CAACAGATCCAGCTCGGCGTCCGCATCGCCCCGCCGCCCGCCGACGATGATGACGATTCG
GTGGCCGCCGCGGTCGAGGACCCCGGCCTCGAAGCGTCACCAGAAACGTCACAAGAAGTC
TCCGACGAGATCGATGACTTTGGCGAAAACGCGCCAAAGTCGCGGCAAAGCTGGCCGACG
CACTTCAAGAAGCGCTCAACCGACGCTGATACCAGCGCTAGCGCCGACGGTACCAGCCTC
AACCGCCGCTACACGTTCGACACCTTCGTCATCGGAGCCTCCAACCGATTCGCGCACGCG


### .tsv file

###### .tsv files gives the summary in tab separated format

In [12]:
head Agy99/Agy99.tsv

locus_tag	ftype	length_bp	gene	EC_number	COG	product
Agy99_00001	CDS	1500	dnaA		COG0593	Chromosomal replication initiator protein DnaA
Agy99_00002	CDS	1209	dnaN		COG0592	Beta sliding clamp
Agy99_00003	CDS	1080	recF		COG1195	DNA replication and repair protein RecF
Agy99_00004	CDS	564				hypothetical protein
Agy99_00005	CDS	2079	gyrB1	5.6.2.2		DNA gyrase subunit B
Agy99_00006	CDS	2520	gyrA1	5.6.2.2		DNA gyrase subunit A
Agy99_00007	tRNA	75				tRNA-Ile(gat)
Agy99_00008	tRNA	74				tRNA-Ala(tgc)
Agy99_00009	CDS	840	tam_1	2.1.1.144		Trans-aconitate 2-methyltransferase


In [38]:
head -n 2 Agy99/Agy99.tsv # extracting just column names and first entry

locus_tag	ftype	length_bp	gene	EC_number	COG	product
Agy99_00001	CDS	1500	dnaA		COG0593	Chromosomal replication initiator protein DnaA


###### It has found a gene that is 1500 base pair long , with Enzyme Commissoin number COG0593